src/main/java/org/venice/beachfront/bfapi/config/GeoServerPKIAuthenticationConfig.java

```diff
@@ -0,0 +1,165 @@
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.ssl.SSLContexts;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+import org.venice.beachfront.bfapi.geoserver.AuthHeaders;
+import org.venice.beachfront.bfapi.geoserver.PKIAuthHeaders;
+
+/**
+ * Configures the API to use PKI authentication when communicating with GeoServer
+ */
+@Configuration
+@Profile({ "pki-geoserver-auth" })
+public class GeoServerPKIAuthenticationConfig extends GeoServerBaseAuthenticationConfig {
+
+	@Value("${http.max.total}")
+	private int httpMaxTotal;
+
+	@Value("${http.max.route}")
+	private int httpMaxRoute;
+
+	@Value("${JKS_FILE}")
+	private String keystoreFileName;
+
+	@Value("${JKS_PASSPHRASE}")
+	private String keystorePassphrase;
+
+	@Value("${PZ_PASSPHRASE}")
+	private String piazzaKeyPassphrase;
+
+	@Bean
+	public RestTemplate restTemplate() throws KeyManagementException, UnrecoverableKeyException, NoSuchAlgorithmException,
+			KeyStoreException, CertificateException, IOException {
+
+		final SSLContext sslContext = SSLContexts.custom().loadKeyMaterial(getStore(), piazzaKeyPassphrase.toCharArray())
+				.loadTrustMaterial(null, new TrustSelfSignedStrategy()).useProtocol("TLS").build();
+		final Registry<CookieSpecProvider> registry = RegistryBuilder.<CookieSpecProvider>create()
+				.register("myspec", new MySpecProvider()).build();
+		final RequestConfig requestConfig = RequestConfig.custom().setCookieSpec("myspec").setCircularRedirectsAllowed(true).build();
+
+		// @formatter:off
+		final HttpClient httpClient = HttpClientBuilder.create()
+				.setDefaultRequestConfig(requestConfig)
+				.setMaxConnTotal(httpMaxTotal)
+				.setSSLContext(sslContext)
+				.setSSLHostnameVerifier(new NoopHostnameVerifier())
+				.setDefaultCookieStore(new BasicCookieStore())
+				.setDefaultCookieSpecRegistry(registry)
+				.setRedirectStrategy(new MyRedirectStrategy())
+				.setMaxConnPerRoute(httpMaxRoute)
+				.setKeepAliveStrategy(getKeepAliveStrategy())
+				.build();
+		// @formatter:off
+
+		final RestTemplate restTemplate = new RestTemplate();
+		restTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory(httpClient));
+
+		final List<HttpMessageConverter<?>> messageConverters = new ArrayList<HttpMessageConverter<?>>();
+		messageConverters.add(new StringHttpMessageConverter());
+		messageConverters.add(new MappingJackson2HttpMessageConverter());
+		restTemplate.setMessageConverters(messageConverters);
```

Comment:
Sorry, just made a small change related to message converters in master.  Might cause you problems when merging... also not entirely sure my change will be necessary after this change because it was related to the tokens request (that request needing to be form-url-encoded).